lib/dom.js

```diff
@@ -2743,6 +2743,65 @@ function needNamespaceDefine(node, isHTML, visibleNamespaces) {
 	}
 	return true;
 }
+
+// match chromium's behavior of ns1, ns2, ns3, etc. (gecko would be "a")
+var GENERATED_PREFIX_PREFIX = 'ns';
+
+/**
+ * Assume a prefix for an attribute that is not in the default namespace and doesn't have a
+ * specified prefix.
+ *
+ * @param {Node} node
+ * @param {{ prefix: string | null; namespace: string | null }[]} visibleNamespaces
+ * @returns
+ * The prefix To use when serializing the attribute node.
+ */
+function assumePrefix(node, visibleNamespaces) {
+	var nodeType = node.nodeType;
+	var prefix = node.prefix || '';
+	var uri = node.namespaceURI;
+
+	if (uri === NAMESPACE.XMLNS) return node.prefix;
+
+	if (uri === '' || uri == null) {
+		// I don't think we should be doing new validation here, but if we wanted to the following
+		// makes sense:
+
+		// if (prefix != '') {
+		//     throw new DOMException(DOMException.NAMESPACE_ERR, 'prefix is non-null and namespace is null');
+		// }
+
+		return null;
```

Comment:
According to the checks this return is not covered by any test, could you add one for it?

Reply:
Awesome, thx